waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/impl/YamlFederatedMetaStoreStorageTest.java

```diff
@@ -89,6 +89,20 @@ public void loadFederation_singleFederation() throws Exception {
     assertThat(storage.get("name"), is(expected));
   }
 
+  @Test (expected = IllegalArgumentException.class)
+  public void loadFederationSamePrefix() throws Exception {
+    File f = dataFolder.getFile("same-prefix.yml");
+    YamlFederatedMetaStoreStorage storage = new YamlFederatedMetaStoreStorage(f.toURI().toString(), configuration);
+    storage.loadFederation();
+  }
+
+  @Test (expected = IllegalArgumentException.class)
+  public void loadTwoEmptyPrefixes() throws Exception {
+    File f = dataFolder.getFile("two-empty-prefixes.yml");
+    YamlFederatedMetaStoreStorage storage = new YamlFederatedMetaStoreStorage(f.toURI().toString(), configuration);
+    storage.loadFederation();
+  }
+
```

Comment:
Is it worth having a test of the "happy day" scenario where have a federated metastore with no prefix and a primary with a prefix? Just to validate that this is allowed and the resultant metastores have the correct prefixes?

Reply:
Yeah, I think it's worth testing that it actually works in a yaml file. I've just added it :)